tests/benchmark/stateful/vector_storage/test_vector_storage.py

```diff
@@ -0,0 +1,207 @@
+    bytecode += Op.ADD                             # counter++
+
+    # Jump back to loop start
+    bytecode += Op.PUSH1(11)                       # Push loop start offset
+    bytecode += Op.JUMP                            # Jump to loop start
+
+    # Exit point (JUMPDEST at offset 31)
+    bytecode += Op.JUMPDEST                        # Exit marker
+    bytecode += Op.STOP                            # Stop execution
+
+    return bytecode
+
+
+@pytest.mark.valid_from("Prague")
+@pytest.mark.stateful  # Mark as stateful instead of benchmark
+@pytest.mark.parametrize("num_slots", [1, 10, 50, 100, 200])
+@pytest.mark.parametrize(
+    "transition_type,start_offset,write_value",
+    [
+        # Write to empty slots (beyond the 500 prefilled)
+        pytest.param(
+            "zero_to_nonzero",
+            PREFILLED_SLOTS,  # Start at slot 500
+            0x1234,
+            id="0_to_nonzero"
+        ),
+
+        # Clear existing slots (write 0 to prefilled slots)
+        pytest.param(
+            "nonzero_to_zero",
+            0,  # Start at slot 0
+            0,  # Write zeros
+            id="nonzero_to_0"
+        ),
+
+        # Update existing slots (change value of prefilled slots)
+        pytest.param(
+            "nonzero_to_nonzero",
+            0,  # Start at slot 0
+            0xBEEFBEEF,  # Different value
+            id="nonzero_to_nonzero"
+        ),
+    ],
+)
+def test_storage_transitions_benchmark(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    num_slots: int,
+    transition_type: str,
+    start_offset: int,
+    write_value: int,
+) -> None:
+    """
+    Benchmark storage operations using a single parametrized contract.
+
+    The contract is pre-deployed with 500 filled slots and accepts
+    parameters via calldata to perform different storage operations:
+
+    1. 0 to nonzero: Writing to cold, empty slots (most expensive)
+    2. nonzero to 0: Clearing existing slots (provides gas refund)
+    3. nonzero to nonzero: Updating warm slots (moderate cost)
+    """
+    sender = pre.fund_eoa(10**18)
+
+    # Pre-fill storage with 500 slots
+    initial_storage = Storage()
+    for i in range(PREFILLED_SLOTS):
+        initial_storage[i] = PREFILLED_VALUE
+
+    # Deploy the optimized contract with prefilled storage
+    storage_contract = pre.deploy_contract(
+        code=create_storage_contract(),
+        storage=initial_storage,
+    )
+
+    # Prepare calldata with parameters
+    calldata = (
+        num_slots.to_bytes(32, 'big') +      # Number of slots to write
+        start_offset.to_bytes(32, 'big') +   # Starting slot index
+        write_value.to_bytes(32, 'big')      # Value to write
+    )
+
+    # Create transaction to call the contract
+    # Use a reasonable gas limit that covers the operation
+    # Each SSTORE costs up to 20,000 gas for cold writes plus overhead
+    gas_limit = 21000 + (num_slots * 50000)  # Base tx cost + storage operations
```

Comment:
I am curious why specifying the gas limit here, will it be affected if we make it the default value?

If necessary, i would suggest to use:


```suggestion
    intrinsic_gas_cost_calc = fork.transaction_intrinsic_cost_calculator()
    gas_limit = intrinsic_gas_cost_calc() + (num_slots * 50000)  # Base tx cost + storage operations
```